src/librbd/crypto/luks/Utils.h

```diff
@@ -0,0 +1,37 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_LUKS_UTILS_H
+#define CEPH_LIBRBD_CRYPTO_LUKS_UTILS_H
+
+#include <stdio.h>
+#include <string>
+#include <unistd.h>
+#include <utility>
+
+#ifndef _GNU_SOURCE
```

Comment:
Nit: might be nicer to move the implementation to a C file to avoid possible pollution of `#define`s.